src/components/tooltip/TrackPower.js

```diff
@@ -0,0 +1,156 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import * as React from 'react';
+import { Localized } from '@fluent/react';
+import memoize from 'memoize-one';
+
+import explicitConnect from 'firefox-profiler/utils/connect';
+import { formatNumber } from 'firefox-profiler/utils/format-numbers';
+import {
+  getCommittedRange,
+  getPreviewSelection,
+  getProfileInterval,
+} from 'firefox-profiler/selectors/profile';
+import { getSampleIndexRangeForSelection } from 'firefox-profiler/profile-logic/profile-data';
+
+import { TooltipDetails, TooltipDetail } from './TooltipDetails';
+
+import type {
+  Counter,
+  Milliseconds,
+  PreviewSelection,
+  StartEndRange,
+} from 'firefox-profiler/types';
+
+import type { ConnectedProps } from 'firefox-profiler/utils/connect';
+
+type OwnProps = {|
+  counter: Counter,
+  counterSampleIndex: number,
+|};
+
+type StateProps = {|
+  interval: Milliseconds,
+  committedRange: StartEndRange,
+  previewSelection: PreviewSelection,
+|};
+
+type Props = ConnectedProps<OwnProps, StateProps, {||}>;
+
+class TooltipTrackPowerImpl extends React.PureComponent<Props> {
+  // This compute the sum of the power in the range. This returns a value in Wh.
+  _computePowerSumForRange(start: Milliseconds, end: Milliseconds): number {
+    const { counter } = this.props;
+    const samples = counter.sampleGroups[0].samples;
+    const [beginIndex, endIndex] = getSampleIndexRangeForSelection(
+      samples,
+      start,
+      end
+    );
+
+    let sum = 0;
+    for (
+      let counterSampleIndex = beginIndex;
+      counterSampleIndex < endIndex;
+      counterSampleIndex++
+    ) {
+      sum += samples.count[counterSampleIndex]; // picowatt-hour;
+    }
+    return sum * 1e-12;
+  }
+
+  _computePowerSumForCommittedRange = memoize(
```

Comment:
I decided to memoize here rather than in redux, to make the component more self contained. I thought this wouldn't be used elsewhere, at least not yet. Maybe we may want to display it elsewhere in the future, but we'd probably need to compute the sum of all tracks too... A problem for another day...